scripts/incremental_validator.py

```
@@ -0,0 +1,322 @@
+
+@dataclass
+class FileValidationCache:
+    file_path: str
+    content_hash: str
+    last_validated: float
+    validation_results: Dict[str, str] = field(default_factory=dict)  # validation_type -> result
+    error_count: int = 0
+
+
+class ValidationCache:
+    def __init__(self, cache_file: Path = None):
```

Ruff (0.14.4)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)